src/python/pants/binaries/binary_tool.py

```diff
@@ -32,6 +35,17 @@ class BinaryToolBase(Subsystem):
   # Subclasses may set this to provide extra register() kwargs for the --version option.
   extra_version_option_kwargs = None
 
+  class InvalidSupportDir(TaskError):
+    """Indicates that the subclass of BinaryToolBase did not set up a valid
```

Comment:
Nit: The first sentence of a docstring should fit on a single line.

I think you can convey this in half the number of words.

Reply:
Also, let's be consistent with the naming of "support_dir" vs. "support_subdir". BinaryUtil uses "supportdir", but it's ok to be inconsistent with that here, since we're going to make BinaryUtil into an implementation detail and not allow direct use of it. But at least within this file let's be consistent.

Reply:
I have removed the mess of joining directory paths and made BinaryTool subclasses go under `bin/`, so I have removed this exception and the naming confusion, as they are no longer needed. I will keep your comments in mind for this and future PRs, though!